src/Repository/UserRepository.php

```diff
@@ -3,27 +3,69 @@
 namespace App\Repository;
 
 use App\Entity\User\User;
+use Exception;
+use App\Exceptions\EntityNotFoundException;
 
-class UserRepository
+
+class UserRepository extends Repository
 {
-    private DBConnexion $dbConnection;
+    protected $table = 'user';
 
-    public function __construct(DBConnexion $dbConnection)
+    public function __construct(\App\Repository\DBConnexion $dbConnection)
     {
-        $this->dbConnection = $dbConnection;
+        try{
+            parent::__construct($dbConnection, $this->table, User::class);
+
+            if(!$this->entity){
+                throw new EntityNotFoundException($this->entity);
+            }
```

Comment:
I don't understand these lines. If you send User::class to your parent construct , this condition will never be true.
For me, "EntityNotFound" it's when we can not found an entity from your db. :s